autobahn_sync/core.py

```diff
@@ -87,6 +89,9 @@ def run(self, url=DEFAULT_AUTOBAHN_ROUTER, realm=DEFAULT_AUTOBAHN_REALM,
         """
         _init_crochet(in_twisted=False)
         self._bootstrap(blocking, url=url, realm=realm, **kwargs)
+        if callback:
```

Comment:
If `blocking=False` (only case callback is interesting) when we hit this line the bootstrap has not been done, so the callback is called too soon !